it/v2/endpoints/CreateHistoricNonFhlUkPiePeriodSummaryControllerISpec.scala

```diff
@@ -0,0 +1,430 @@
+      |    "travelCosts": 645.56999999999999999999999,
+      |    "residentialFinancialCostsCarriedForward": 672.34999999999999999999999,
+      |    "residentialFinancialCost": 1000.45999999999999999999999,
+      |    "rentARoom": {
+      |      "amountClaimed": 545.9999999999999999999999
+      |    }
+      |  }
+      |}
+      |""".stripMargin
+  )
+
+  val bothExpensesSuppliedRequestJson: JsValue = Json.parse(
+    """{
+      | "fromDate": "2019-03-11",
+      | "toDate": "2020-04-23",
+      |   "income": {
+      |   "periodAmount": 123.45,
+      |   "premiumsOfLeaseGrant": 2355.45,
+      |   "reversePremiums": 454.56,
+      |   "otherIncome": 567.89,
+      |   "taxDeducted": 234.53,
+      |   "rentARoom": {
+      |      "rentsReceived": 567.56
+      |    }
+      |   },
+      |  "expenses":{
+      |    "premisesRunningCosts": 567.53,
+      |    "repairsAndMaintenance": 324.65,
+      |    "financialCosts": 453.56,
+      |    "professionalFees": 535.78,
+      |    "costOfServices": 678.34,
+      |    "other": 682.34,
+      |    "travelCosts": 645.56,
+      |    "residentialFinancialCostsCarriedForward": 672.34,
+      |    "residentialFinancialCost": 1000.45,
+      |    "rentARoom": {
+      |      "amountClaimed": 545.9
+      |    },
+      |    "consolidatedExpenses": 235.78
+      |  }
+      |}""".stripMargin
+  )
+
+  val toDateBeforeFromDateRequestJson: JsValue = Json.parse(
+    """{
+      | "fromDate": "2020-03-11",
+      | "toDate": "2019-04-23",
+      |   "income": {
+      |   "periodAmount": 123.45,
+      |   "premiumsOfLeaseGrant": 2355.45,
+      |   "reversePremiums": 454.56,
+      |   "otherIncome": 567.89,
+      |   "taxDeducted": 234.53,
+      |   "rentARoom": {
+      |      "rentsReceived": 567.56
+      |    }
+      |   },
+      |  "expenses":{
+      |    "premisesRunningCosts": 567.53,
+      |    "repairsAndMaintenance": 324.65,
+      |    "financialCosts": 453.56,
+      |    "professionalFees": 535.78,
+      |    "costOfServices": 678.34,
+      |    "other": 682.34,
+      |    "travelCosts": 645.56,
+      |    "residentialFinancialCostsCarriedForward": 672.34,
+      |    "residentialFinancialCost": 1000.45,
+      |    "rentARoom": {
+      |      "amountClaimed": 545.9
+      |    }
+      |  }
+      |}
+      |""".stripMargin
+  )
+
+  val missingFromAndToDateRequestJson: JsValue = Json.parse(
+    """{
+      |    "income": {
+      |        "periodAmount": 123.45,
+      |        "premiumsOfLeaseGrant": 2355.45,
+      |        "reversePremiums": 454.56,
+      |        "otherIncome": 567.89,
+      |        "taxDeducted": 234.53,
+      |        "rentARoom": {
+      |           "rentsReceived": 567.56
+      |         }
+      |        },
+      |       "expenses":{
+      |          "consolidatedExpenses": 235.78
+      |     }
+      |}""".stripMargin
+  )
+
+  val allInvalidValueRequestError: MtdError = ValueFormatError.copy(
```

Comment:
You could just have a single bad value to prove that `ValueFormatError` is picked up. (Testing that we pick up multiple errors is more at home in a validation unit test than an integration test - and I suspect we already do this.) Then you can move this to an in-line value at the point where it is used.

Reply:
Simplified + updated